File: app/src/main/java/com/dertyp7214/changeloglibexampleapp/MainActivity.java
package com.dertyp7214.changeloglibexampleapp;

import android.annotation.SuppressLint;
import android.graphics.Color;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Html;
import android.util.Log;
import android.webkit.WebView;
import android.widget.TextView;

import com.dertyp7214.changelogs.ChangeLog;
import com.dertyp7214.changelogs.Version;

public class MainActivity extends AppCompatActivity {

    @SuppressLint("SetJavaScriptEnabled")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ChangeLog changeLog = new ChangeLog.Builder(this)
                .addVersion(new Version.Builder(this)
                        .setVersionName("2.3")
                        .setVersionCode("234")
                        .addChange(new Version.Change(Version.Change.ChangeType.REMOVE, "Something"))
                        .build())
                .addVersion(new Version.Builder(this)
                        .setVersionName("2.2")
                        .setVersionCode("223")
                        .addChange(new Version.Change(Version.Change.ChangeType.ADD, "Something"))
                        .build())
                .setLinkColor(Color.GREEN)
                .build();
        changeLog.buildDialog("Changes").showDialog();

    }
}

File: app/src/main/java/com/dertyp7214/changeloglibexampleapp/MainActivity.kt
package com.dertyp7214.changeloglibexampleapp

import android.annotation.SuppressLint
import android.app.Activity
import android.os.Bundle
import android.util.Log
import com.dertyp7214.changelogs.Changelog

class MainActivity : Activity() {

    @SuppressLint("SetJavaScriptEnabled")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val changelog = Changelog(this) {
            addVersion {
                versionName = "3"
                versionCode = "300"

                addChange {
                    type = add()
                    change = "Test"
                }
                addChange {
                    type = fix()
                    addLink("Test Link", "https://google.com")
                }
            }
            addVersion {
                versionName = "2"
                versionCode = "200"

                addChange {
                    type = fix()
                    change = "Fixed stuff"
                }
            }
            onClose {
                Log.d("REEEE", "$it")
            }
        }

        if (changelog.showDialogOnVersionChange())
            Log.d("DIALOGSHOWN", "TRUE")
        else {
            Log.d("DIALOGSHOWN", "FALSE")
            changelog.show()
        }
    }
}
